src/common/Surface.js

```diff
@@ -130,13 +137,45 @@
             .display(this.showTitle())
             .render()
         ;
+
+        var surfaceButtons = element.selectAll(".surface-button").data(this.buttonAnnotations());
+        surfaceButtons.enter().append("tspan")
+            .each(function (button, idx) {
+                context._surfaceButtons[idx] = new Icon()
+                .faChar(button["char"])
+                .diameter(button["diamter"])
+                .image_colorFill(button["image_colorFill"])
+                .shape_colorFill(button["shape_colorFill"])
+                .shape_colorStroke(button["shape_colorStroke"])
+                .paddingPercent(button["padding"])
+                .shape(button["shape"])
+                .target(domNode)
+                .display(context.showTitle() && context.showIcon())
+                .render(function(widget) {
+                    widget.element().style("cursor","pointer");
+                    widget.element().classed("surface-button", true);
+                    widget.click = function(d) { context.click(d,widget); };
+                });
+            })
+
+        ;
+        surfaceButtons.exit()
+            .each(function (d, idx) {
+                var element = d3.select(this);
+                delete context._surfaceButtons[idx];
+                element.remove();
+            })
+        ;
+
+
+        var buttonClientSize = this.showIcon() ? this._icon.getBBox(true) : {width:0, height: 0};
```

Comment:
This is incorrect - it should loop through your array of _surfaceButtons and find the max height.